src/components/side_panel/chart/building_blocks/title/title.ts

```diff
@@ -1,18 +1,60 @@
-import { Component } from "@odoo/owl";
+import { Component, useState } from "@odoo/owl";
 import { SpreadsheetChildEnv } from "../../../../../types";
+import { SelectionInput } from "../../../../selection_input/selection_input";
+import { Checkbox } from "../../../components/checkbox/checkbox";
 import { Section } from "../../../components/section/section";
 
 interface Props {
   title: string;
   update: (title: string) => void;
+  isTitleCellReferenced: boolean;
+  updateIsTitleCellReferenced: (isTitleCellReferenced: boolean) => void;
+  titleCellReference: string;
+  updateTitleCellReference: (titleCellReference: string) => void;
+}
+
+interface State {
+  dataRange: string;
 }
 
 export class ChartTitle extends Component<Props, SpreadsheetChildEnv> {
   static template = "o-spreadsheet.ChartTitle";
-  static components = { Section };
-  static props = { title: String, update: Function };
+  static components = { Section, Checkbox, SelectionInput };
+  static props = {
+    title: String,
+    update: Function,
+    isTitleCellReferenced: Boolean,
+    updateIsTitleCellReferenced: Function,
+    titleCellReference: String,
+    updateTitleCellReference: Function,
+  };
+
+  private state: State = useState({
+    dataRange: this.props.titleCellReference || "",
+  });
 
   updateTitle(ev: InputEvent) {
     this.props.update((ev.target as HTMLInputElement).value);
   }
+
+  handleSelectionChange(ranges: string[]) {
+    this.state.dataRange = ranges[0];
+  }
+
+  updateDataRange() {
```

Comment:
```suggestion
  updateTitleReference() {
```
of updateTitleRangs, or something like that. I don't think the name "data" is fine for a title :)